### ISSUE TITLE
No data in Default metric

### ISSUE THREAD

**Issue by Sundrops** [microsoft/nni#1939]
I want to run comands of trial in a docker container. And I found the metics in /mnt/share/alex/nni/experiment/xxxx/trials/xxxxx/.nni/metrics after I run `nnictl create --config config.yaml`. But, they were not shown in Default metric(WebUI). 
```yaml
authorName: default
experimentName: example_mnist_pytorch
trialConcurrency: 2
maxExecDuration: 1h
maxTrialNum: 40
trainingServicePlatform: local
searchSpacePath: search_space.json
useAnnotation: false
logDir: /mnt/share/alex/nni/experiment
tuner:
  builtinTunerName: TPE
  classArgs:
    optimize_mode: maximize
trial:
# /mnt/share is a share dir(local and container both can access it)
  command: run my docker xxxx -c "python /mnt/share/alex/projects/nni/examples/mnist-pytorch/mnist.py
  codeDir: .
  gpuNum: 1
```
```
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 0, "value": 31.56}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 1, "value": 58.88}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 2, "value": 68.79}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 3, "value": 71.58}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 4, "value": 77.17}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 5, "value": 81.07}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 6, "value": 83.97}
ME000098{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 7, "value": 85.91}
ME000097{"parameter_id": 2, "trial_job_id": "atKaa", "type": "PERIODICAL", "sequence": 8, "value": 87.1}
ME000093{"parameter_id": 2, "trial_job_id": "atKaa", "type": "FINAL", "sequence": 0, "value": 88.04}

```
![image](https://user-images.githubusercontent.com/14829222/72073963-f5e11280-332b-11ea-9bee-ee935172a0f8.png)

**Comment by Sundrops** [comment]
Maybe the ip of docker container is different from host computer's.

**Comment by SparkSnail** [comment]
Hi @Sundrops, your docker container runs in remote server, how do you share a folder with nniManager machine? 
If you want to run trial jobs in remote docker, you could use `remote` mode in NNI, set the remote container as remote machine, and use ssh to connect the docker.

**Comment by Sundrops** [comment]
@SparkSnail 
Thanks for your reply. In my company, my machine that runs nni shares the dir with remote servers. I can not use remote mode to run  trial jobs, because I can not directly connect the remote docker through ssh. The pipeline is: I run nni in my machine, and the trial command includes applying for a remote docker in a queue and runing the training task in remote docker.  After the training task launched, the nni server in my machine cannot show the default metics though the metics has been in the log file `.nni/.metrics`

**Comment by Sundrops** [comment]
@SparkSnail 
As  https://github.com/microsoft/nni/issues/266 said, this problem is induced by that tail-stream package does not work in remote docker container. metrics file is changed, but tail-stream does not notice that.  The nni version I used is 1.3,

**Comment by SparkSnail** [comment]
@ Sundrops, I see. There might be some error when using stream to watch file change. The related 
 file stream code is in https://github.com/microsoft/nni/blob/master/src/nni_manager/training_service/local/localTrainingService.ts#L529.
I  wonder how do you share a folder with remote container, so I could reproduce the problem. Are you using NFS mount?

**Comment by Sundrops** [comment]
@SparkSnail , Yes, I use NFS mount. My machine and the docker runing on remote server both can access the dir `/mnt/machine_#id`.

**Comment by Sundrops** [comment]
@SparkSnail Can you reproduce the problem?

**Comment by SparkSnail** [comment]
@Sundrops How do you start remote docker by `run my docker xxxx` command, since the trial runs on local machine?

**Comment by Sundrops** [comment]
@SparkSnail I used a tool(xxx run --image docker_image_name --command 'sh train.sh') provided by my company to submit a job to cluster(k8s).

**Comment by SparkSnail** [comment]
NNI support running jobs in K8S clusters by KubeflowTrainingService or FramewrokControllerTrainingService, have you ever tried that?

**Comment by Sundrops** [comment]
@SparkSnail I don't have permission to modify the config of customized k8s in my company. Maybe the problem is related to docker version and kernel version.

**Comment by zym604** [comment]
@SparkSnail
Same problem:
![Capture](https://user-images.githubusercontent.com/9557009/73772825-f31ce600-474e-11ea-843d-89737946ab8d.PNG)
![Capture2](https://user-images.githubusercontent.com/9557009/73772830-f6b06d00-474e-11ea-9076-b94607bc7df6.PNG)
System: Windows 10 Pro
code: https://github.com/microsoft/nni/blob/master/examples/trials/mnist-pytorch/config_windows.yml
(change `gpuNum: 0` to `gpuNum: 1`)
GPU: NVIDIA GeForce GTX 1070
pytorch version: 1.4.0
nni version: 1.3
cudatoolkit version: 10.1.243

Attached are the log files:
https://drive.google.com/file/d/16h7-y3wFZORyn4LKcyTrIZzLptFtWZQ1/view?usp=sharing

**Comment by Sundrops** [comment]
@SparkSnail I have fixed this bug. Make the machine ID launched the nni same as the the machine ID in the logdir: /mnt/ID(nfs)

**Comment by Sundrops** [comment]
@zym604 Maybe you didn't report final result. My problem is different from yours.

**Comment by zym604** [comment]
@Sundrops 
I did.
```    for epoch in range(1, args['epochs'] + 1):
        train(args, model, device, train_loader, optimizer, epoch)
        test_acc = test(args, model, device, test_loader)

        if epoch < args['epochs']:
            # report intermediate result
            nni.report_intermediate_result(test_acc)
            logger.debug('test accuracy %g', test_acc)
            logger.debug('Pipe send intermediate result done.')
        else:
            # report final result
            nni.report_final_result(test_acc)
            logger.debug('Final result is %g', test_acc)
            logger.debug('Send final result done.')

I use the default pytorch example on NNI.
https://github.com/microsoft/nni/blob/master/examples/trials/mnist-pytorch/mnist.py

**Comment by Sundrops** [comment]
You can get LATEST result while I get nothing in Default metric, so my problem is different from yours. Which mode do you use? local? remote?

**Comment by zym604** [comment]
@Sundrops local

**Comment by Sundrops** [comment]
@zym604 You check whether the `.nni/metrics` file contains `ME000107{"parameter_id": xxxx, "trial_job_id": "xxxx", "type": "**FINAL**", "sequence": xxxx, "value": xxxx}`
